lib/ansible/modules/storage/netapp/na_ontap_snapshot.py

```diff
@@ -0,0 +1,320 @@
+#!/usr/bin/python
+
+# (c) 2018, NetApp, Inc
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+module: na_ontap_snapshot
+short_description: Manage NetApp Sanpshots
+extends_documentation_fragment:
+    - netapp.na_ontap
+version_added: '2.6'
+author:
+- Chris Archibald (carchi@netapp.com), Kevin Hutton (khutton@netapp.com)
+description:
+- Create/Modify/Delete Ontap snapshots
+options:
+  state:
+    description:
+    - If you want to create/modify a snapshot, or delete it.
+    choices: ['present', 'absent']
+    default: present
+  snapshot:
+    description:
+    - Name of the snapshot to be created.
+    - The maximum string length is 256 characters.
+    required: true
+  volume:
+    description:
+    - Name of the volume on which the snapshot is to be created.
+    required: true
+  async_bool:
```

Comment:
odd parameter name, maybe turn it around to `wait_for_snapshot_completion` (default true) or `create_async` or something?

Reply:
create_asynchronously

Reply:
So we tried to keep the same names we used in ontap. In Ontap we use async, with Python 3 async is now a reserved word. I could change this to asynchronously or something similar.